### PR TITLE
Adds unit tests for NumberQuestion

### LINKED ISSUE #16: Render integer questions
Questions involving integers (e.g. "What is your annual income?") should have:

- [x] The question title
- [x] A question hint
- [x] A textbox with integer validation
- [x] Validation and error text logic to require non-negative integers (>=0)
- [x] Use HTML5 validation to use numeric keypad
- [x] Testing

### PR DESCRIPTION
**Pull request by kebrahim**
Works on #16 

Covered the following test cases:
- Rendering with question text/hint
- Rendering with question text without hint
- Rendering with response pulled from Redux
- Updating input with valid response, including Redux
- Ignoring invalid response (non-integer)
- Validation (I'm assuming Michelle will cover this in her PR)
- a11y (I'll do this in a subsequent PR)

### CODE REVIEW

File: client/src/components/NumberQuestion.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    fireEvent.change(numberInput, { target: { value: "hi there" } });
+
+    // Verify the invalid input is not saved in UI or Redux
+    expect(
+      getByTestId("integer-question-HOUSEHOLD_SIZE").children[0].value
+    ).toEqual("");
+    expect(store.getActions()).toHaveLength(0);
+  });
+
+  // TODO(mscharfstein): Add test for validation
```
Comment (mscharfstein) [inline client/src/components/NumberQuestion.test.js:134]: the above test is already a test for validation i think..

File: client/src/components/NumberQuestion.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      if (key != "catalog:ADULTS_65_PLUS.hint") {
+        return key;
+      } else {
+        return null;
+      }
+    },
+  }),
+}));
+
+const middlewares = [thunk];
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:21]: What do you use thunk for?

File: client/src/components/NumberQuestion.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    const questionKey = "HOUSEHOLD_SIZE";
+    const store = mockStore({
+      questions: mockQuestions,
+      responses: {},
+    });
+    const { getByText, getByTestId } = render(
+      <Provider store={store}>
+        <NumberQuestion questionKey={questionKey} />
+      </Provider>
+    );
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:36]: This is a lot of repetitive overhead for each test. See my PR #94 for an alternative.

File: client/src/components/NumberQuestion.test.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    expect(getByText("catalog:HOUSEHOLD_SIZE.text")).toBeInTheDocument();
+    expect(getByText("catalog:HOUSEHOLD_SIZE.hint")).toBeInTheDocument();
+    expect(
+      getByTestId("integer-question-HOUSEHOLD_SIZE").children[0].value
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:40]: I don't think you need a testid here. Maybe:
```js
getByLabelText("catalog:HOUSEHOLD_SIZE.text").toHaveValue("")
```
[toHaveValue](https://github.com/testing-library/jest-dom#tohavevalue)

File: client/src/components/NumberQuestion.test.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    ).toEqual("");
+  });
+
+  it("Hides hint label when hint not specified", () => {
+    const questionKey = "ADULTS_65_PLUS";
+    const store = mockStore({
+      questions: mockQuestions,
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:47]: Not convinced that using all this real data as mock is necessarily clearer in these simple cases.

File: client/src/components/NumberQuestion.test.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      responses: {},
+    });
+    const { getByText, queryByText, getByTestId } = render(
+      <Provider store={store}>
+        <NumberQuestion questionKey={questionKey} />
+      </Provider>
+    );
+    expect(getByText("catalog:ADULTS_65_PLUS.text")).toBeInTheDocument();
+    expect(queryByText("catalog:ADULTS_65_PLUS.hint")).toBeNull();
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:56]: Better would be:
```js
expect(queryByText("...")).not.toBeInTheDocument()
//or
expect(queryByText("...")).not.toBeVisible()
```

File: client/src/components/NumberQuestion.test.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <NumberQuestion questionKey={questionKey} />
+      </Provider>
+    );
+
+    // Update number input to a valid integer
+    const numberInput = getByTestId("integer-question-HOUSEHOLD_SIZE")
+      .children[0];
+    expect(numberInput.value).toEqual("");
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:95]: See above. Better to use the .toHaveValue matcher.

File: client/src/components/NumberQuestion.test.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    fireEvent.change(numberInput, { target: { value: "3" } });
+    expect(
+      getByTestId("integer-question-HOUSEHOLD_SIZE").children[0].value
+    ).toEqual("3");
```
Comment (marcja) [inline client/src/components/NumberQuestion.test.js:99]: What is the point of this assertion? On the line above, you're changing the state of the HTML element and immediately validating that the HTML element still has that value. How could this fail?

File: client/src/components/NumberQuestion.test.js (hunk 1 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: good point, removed TODO

File: client/src/components/NumberQuestion.test.js (hunk 2 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: removed

File: client/src/components/NumberQuestion.test.js (hunk 3 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: thanks for the tips. i cleaned it up tremendously, using both yours & ethan's PRs as guides. it looks much better now :)

File: client/src/components/NumberQuestion.test.js (hunk 8 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: removed

**Comment by kebrahim** [comment]
> This all looks good, could you also add tests checking that negative & decimal values show the error state?

done

File: client/src/components/NumberQuestion.test.js (hunk 6 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: done

File: client/src/components/NumberQuestion.test.js (hunk 7 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: done

File: client/src/components/NumberQuestion.test.js (hunk 4 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: done

File: client/src/components/NumberQuestion.test.js (hunk 5 above)
Comment (kebrahim) [reply client/src/components/NumberQuestion.test.js]: removed
